docs/src/arrays.md

```diff
@@ -0,0 +1,219 @@
+
+$$ds = \Re\langle \overline{C}, dC \rangle = \Re\left(\sum_{i,\dots,j} \overline{C}_{i,\dots,j}^* dC_{i,\dots,j}\right),$$
+
+where $\cdot^*$ is the complex conjugate, and $\Re(x)$ is the real part of $x$ (i.e. `real(x)`).
+
+For matrices and vectors, we can write this as
+
+$$ds = \Re(\operatorname{tr}(\overline{C}^H\ dC)),$$
+
+where $\operatorname{tr}$ is the `trace`.
+
+Plugging in the identity for $dC$, we get
+
+$$ds = \Re\left(\operatorname{tr}\left(\overline{C}^H \frac{\partial f}{\partial A} dA + \overline{C}^H\frac{\partial f}{\partial B} dB\right)\right) = \Re\left(\operatorname{tr}\left(\overline{C}^H \frac{\partial f}{\partial A} dA\right)\right) + \Re\left(\operatorname{tr}\left(\overline{C}^H\frac{\partial f}{\partial B} dB\right)\right)$$
+
+By applying the same definition of $ds$ to the intermediates $A$ and $B$, we get
+
+$$ds = \Re\left(\operatorname{tr}(\overline{A}^H\ dA)\right) + \Re\left(\operatorname{tr}(\overline{B}^H\ dB)\right)$$
+
+Combining these two identities and solving for $\overline{A}$ and $\overline{B}$ gives us
+
+$$\overline{A} = (\overline{A}^H)^H = \left(\overline{C}^H \frac{\partial f}{\partial A}\right)^H = \left(\frac{\partial f}{\partial A}\right)^H \overline{C}$$
+$$\overline{B} = \left(\frac{\partial f}{\partial B}\right)^H \overline{C}$$
+
+Giles' method for deriving pullback functions is to first derive the differential identity (i.e. pushforward) using the above approach, then pre-multiply by $\overline{C}^H$, and take the trace.
+Subsequently, manipulate into this form and solve for the adjoint derivatives of the inputs.
+Several properties of `trace` make this easier:
+
+$$\operatorname{tr}(A+B) = \operatorname{tr}(A) + \operatorname{tr}(B)$$
+$$\operatorname{tr}(A^T) = \operatorname{tr}(A)$$
+$$\operatorname{tr}(A^H) = \operatorname{tr}(A)^*$$
+$$\operatorname{tr}(AB) = \operatorname{tr}(BA)$$
+
+
+!!! note
+    Our method is identical to Giles' approach, except we have replace the transpose with the conjugate transpose, and we have added the constraint that the inner product be real.
+    This produces the correct pullbacks for arrays with complex entries.
+
+Here are a few examples.
+
+### Matrix multiplication
+
+We above derived
+
+$$dC = dA\ B + A\ dB$$
+
+We now multiply by $\overline{C}^H$ and take the real trace:
+
+$$ds = \Re\left(\operatorname{tr}(\overline{C}^H dC)\right) = \Re\left(\operatorname{tr}(\overline{C}^H dA\ B)\right) + \Re\left(\operatorname{tr}(\overline{C}^H A\ dB)\right)$$
+
+We use the `trace` identities to manipulate the $dA$ expression:
+
+$$ds = \Re\left(\operatorname{tr}(B \overline{C}^H dA)\right) + \Re\left(\operatorname{tr}(\overline{C}^H A\ dB)\right)$$
+
+That's it!
+The expression is in the desired form to solve for the adjoints:
+
+$$\overline{A} = (B \overline{C}^H)^H = \overline{C} B^H$$
+$$\overline{B} = (\overline{C}^H A)^H = A^H \overline{C}$$
+
+Using our notation conventions, we would implement the `rrule` as
```

Comment:
Consider `Using ChainRules' notation`, or something like that to distinguish from the notation in this doc.